packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -88,20 +102,28 @@ export const MatterportIntegration: React.FC = () => {
   const onMatterportModelIdChange = useCallback(
     (event) => {
       const value = event.detail.value;
-      if (value !== matterportModelId) {
-        setMatterportModelId(value);
+      if (value !== matterportModelId.current) {
+        matterportModelId.current = value;
         setDirty(true);
       }
     },
-    [setMatterportModelId, setDirty],
+    [setDirty],
   );
 
   useEffect(() => {
     if (dirty) {
-      setSceneProperty(KnownSceneProperty.MatterportModelId, matterportModelId !== '' ? matterportModelId : undefined);
+      setSceneProperty(
+        KnownSceneProperty.MatterportModelId,
+        matterportModelId.current !== '' ? matterportModelId.current : undefined,
+      );
       setDirty(false);
+      setMatterportViewer(
+        matterportModelId.current !== undefined &&
+          matterportModelId.current !== '' &&
+          selectedConnectionName !== OPTIONS_PLACEHOLDER_VALUE,
+      );
     }
-  }, [matterportModelId]);
+  }, [matterportModelId.current]);
```

Comment:
why does matter port model id need to be a ref? Refs are typically used for things that you _don't_ want to trigger re-renders, but it looks like we do want it to trigger a re-render? Using lots of hooks and refs, and dirty flags complicates things and should be avoided if possible

https://react.dev/learn/referencing-values-with-refs

Reply:
Not using useRef anymore